### PR TITLE
Enhance/parser unit test

### PR DESCRIPTION
**Pull request by bsmmoon**
Cleaned up messy test cases to more logical sequence of tests.

It closes #111

### CODE REVIEW

**Comment by starkipraggy** [comment]
The testing classes need more specific names. It is not immediately apparent what exactly is being tested by its name. It will be useful when tests start failing and you can tell from the test name what might possibly have gone wrong.

Also, consider moving the tests from the bottom to the top. The tests near the bottom are more like unit tests that rigourously test a specific parser class while the tests on top tests the parent class which tests the child classes in tandem which is more like a mini-integration test.

**Comment by bsmmoon** [comment]
@starkipraggy OK I tried to rename them much more descriptive way. Also removed many redundant tests and order of the things so I hope it is more readable now.

File: DoLahTest/ParserUnitTest.cpp [hunk 1]
```diff
@@ -62,181 +73,307 @@ namespace DoLahTest {
...
+        try {
+            DoLah::CommandParser::parse(input);
+            Assert::IsTrue(false);
+        } catch (std::invalid_argument e) {
+            Assert::AreEqual(TOO_LITTLE_ARGUMENTS_MESSAGE, (std::string) e.what());
         }
     }
 
-    TEST_METHOD(parseEditTest2) {
+    TEST_METHOD(WellDeginedEditCommand) {
```
Comment (starkipraggy) [inline DoLahTest/ParserUnitTest.cpp]: WellDefinedEditCommand

File: DoLahTest/ParserUnitTest.cpp [hunk 2]
```diff
@@ -53,6 +56,14 @@ namespace DoLahTest {
         }
     }
 
+    std::string UNHANDLED_COMMAND_MESSAGE = "Command not handled";
```
Comment (starkipraggy) [inline DoLahTest/ParserUnitTest.cpp]: You may want to begin this section with a comment like "Unit Test for individual Parser components". Same for the above section, "Unit Test for main Parser class".

Comment (bsmmoon) [inline DoLahTest/ParserUnitTest.cpp]: K I added

**Pull request merged**